src/content/en/fundamentals/engage-and-retain/push-notifications/sending-messages.markdown

```diff
@@ -3,93 +3,343 @@ layout: shared/narrow
 title: "Sending messages"
 description: "There are two servers involved in sending a message: your server and a third party messaging server. You keep track of who to send messages to. The third party server handles the routing."
 published_on: 2016-07-01
-updated_on: 2016-07-01
+updated_on: 2016-08-05
 order: 30
 translation_priority: 0
 authors:
   - josephmedley
 ---
 
-<p class="intro"> It's easy to assume that sending messages must involve
-building a complex messaging system on the server. That's not quite right. There
-are actually two servers involved in sending a message: your server and a third
-party messaging server. You keep track of two things: recipients and recipient-
-specific endpoints on the messaging server. The messaging server handles the
-routing. </p>
+<p class="intro">
+It's easy to assume that sending messages must involve building a complex 
+messaging system on the server. That's not quite right. There are actually two 
+servers involved in sending a message: your server and a third party messaging 
+server. You keep track of two things: recipients and recipient-specific 
+endpoints on the messaging server. The messaging server handles the routing.
+</p>
 
 {% include shared/toc.liquid %}
 
-## Anatomy of a subscription object {#subscription-anatomy}
+## The broader context
 
-In [the last section](permissions-subscriptions#passing-subscription) we said that a subscription object must be stringified and
-passed to the server. The data the server gets looks like this:
+We've seen how to subscribe to push within a web application. That process included capturing a public key, called the `applicationServerKey` in the subscription API.  
 
-{% highlight json %}
-{
-  "endpoint": "https://example.com/push-service/send/dbDqU8xX10w:APA91b...",
-  "keys": {
-    "auth": "qLAYRzG9TnUwbprns6H2Ew==",
-    "p256dh": "BILXd-c1-zuEQYXH\_tc3qmLq52cggfqqTr\_ZclwqYl6A7-RX2J0NG3icsw..."
-  }
+The following diagram shows the order of operations.
+
+![Sending a message](images/push-flow.gif)
+
+1. A device downloads your web app containing an already created publicKey, referred to in scripts as the `applicationServerKey`. Your web app installs a service worker.
+1. During the subscription flow the browser posts a request to the messaging server to create a new subscription.
+
+   Note: You don't need to know the URL of the message server. Each browser vender manages it's own message server for its browser.
+
+1. The messaging server returns 201 to indicate that the subscription was successful.
+1. After processing the subscription flow, your app passes a subscription object back to your app server. 
```

Comment:
This would be easier to say something less specific.

"1.) The browser contacts the web push service to get a subscription and returns it to.
1.) You send the subscription to your applicationServer."